include/eve/module/real/core/function/regular/generic/load.hpp

```diff
@@ -240,69 +139,33 @@ namespace eve::detail
 #define DISABLE_SANITIZERS
 #endif
 
-  // making unsafe(load) call intinsics requires a lot of work and is not very portable
+  // making unsafe(load) call intrinsics requires a lot of work and is not very portable
   // due to what called functions it will affect.
   // Since the unsafe is mostly used in corner cases, will do it scalar in asan mode.
-  template<relative_conditional_expr C, typename Ptr, typename Cardinal>
-  requires(sanitizers_are_on)
-  DISABLE_SANITIZERS auto load_(EVE_SUPPORTS(cpu_), C const &cond, unsafe_type, Ptr ptr, Cardinal const&) noexcept
-                  -> decltype( eve::load[cond](ptr, Cardinal{}))
-  {
-    using e_t = std::remove_cvref_t<decltype(*ptr)>;
-    using r_t = as_wide_t< e_t, typename Cardinal::type >;
-
-    r_t that;
-
-    auto offset = cond.offset( as_<r_t>{} );
-    auto count  = cond.count( as_<r_t>{} );
-
-    for (std::ptrdiff_t i = offset; i != count + offset; ++i) that.set(i, ptr[i]);
-
-    return that;
-  }
-
-  template<relative_conditional_expr C,typename Ptr>
-  requires(sanitizers_are_on)
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), C const &cond, unsafe_type, Ptr ptr) noexcept
-              ->  decltype(eve::load[cond](ptr))
-  {
-    using T = std::remove_cvref_t<decltype(*ptr)>;
-    return eve::unsafe(eve::load[cond])(ptr, expected_cardinal_t<T>{});
-  }
-
-  template<typename Ptr, typename Cardinal>
-  requires(sanitizers_are_on)
-  DISABLE_SANITIZERS auto load_(EVE_SUPPORTS(cpu_), unsafe_type, Ptr ptr, Cardinal const& N) noexcept
-                  -> decltype(eve::load(ptr, N))
+  template<relative_conditional_expr C, data_source Ptr, typename Pack>
+  DISABLE_SANITIZERS Pack load_ ( EVE_SUPPORTS(cpu_)
+                                , C const& cond, unsafe_type const&
+                                , eve::as_<Pack> const& tgt, Ptr ptr
+                                ) noexcept
   {
-    return eve::unsafe(eve::load[eve::ignore_none])(ptr, N);
-  }
+    if constexpr(sanitizers_are_on && !std::same_as<C,ignore_none_> )
```

Comment:
Only sanitizers are on.

Ignore none will be ignore none in strlen case as well. But it needs to be unsafe.

Reply:
this was changed due to this remark of yours:

```
and not `ignore_none`
```
_Originally posted by @DenisYaroshevskiy in https://github.com/jfalcou/eve/pull/783#discussion_r652838314_